### PR TITLE
total upload and download

### PR DESCRIPTION
**Pull request by eylles**
i hacked together an ugly total upload and download indicator for the popup, it uses a helper shell script, opened the pr cuz the feature is useful enough as is, will rewrite the helper script in lua.

### CODE REVIEW

**Comment by pltanton** [comment]
Good idea! 

Yes, script is pretty simple, I'm sure you can easily rewrite it on lua for consistency and get rid of Makefile. :) Also touching of global environment not a good idea, because not every setup allows it (f.e. NixOs/Guix systems as well as some RO systems on usb dongles).

Also It will be great if this metrics will be optional, just like `popup_signal` 
https://github.com/pltanton/net_widgets/blob/538b11cd6ecac289d9bc5b556e6f66452c75343e/wireless.lua#L148
https://github.com/pltanton/net_widgets/blob/538b11cd6ecac289d9bc5b556e6f66452c75343e/wireless.lua#L76

**Comment by eylles** [comment]
was working on another pr, now i'm back to this one, metrics are an option.
btw i'm not sure if once i rewrite the script add it as it's own script or as a function inside wireless.lua
personally i think leaving it as it's own .lua script and just require it if metrics are set to true is the way to go.

**Comment by eylles** [comment]
at the end i made it into a function

netstats

File: wireless.lua [hunk 1]
```diff
@@ -148,14 +147,24 @@ local function worker(args)
...
+
+						metrics_down = ""
+						metrics_up = ""
+						if popup_metrics then
+							local tdown    = os.capture("ntotal "..interface.." d")
+							local tup    = os.capture("ntotal "..interface.." u")
+							metrics_down = "├DOWN:\t\t"..tdown.."\n"
+              metrics_up = "├UP:\t\t"..tup.."\n"
+						end
+
```
Comment (pltanton) [inline wireless.lua:179]: Replace tabs with spaces, please, because there is no tabs elsewhere :)

File: wireless.lua [hunk 2]
```diff
@@ -47,6 +47,39 @@ local function draw_signal(level)
...
+    end
+  end
+	local count = 0
+	local stat = tonumber(f)
+	while (stat > 999) do
+		stat = (stat / 1024)
+		count = count + 1
+	end
+	result = (round(stat,2) .." "..prefix[count].."B")
+	return result
```
Comment (pltanton) [inline wireless.lua:81]: Also tabs

File: wireless.lua [hunk 3]
```diff
@@ -47,6 +47,39 @@ local function draw_signal(level)
...
+end
+  if (which == "d") then
+    f = readAll("/sys/class/net/" .. card .. "/statistics/rx_bytes")
+  else if (which == "u") then
+    f = readAll("/sys/class/net/" .. card .. "/statistics/tx_bytes")
+    end
+  end
+	local count = 0
+	local stat = tonumber(f)
+	while (stat > 999) do
```
Comment (pltanton) [inline wireless.lua:76]: Why not 1024?

Comment (eylles) [reply wireless.lua]: carried it over from the ugly hacky script, fixed now

File: wireless.lua (hunk 2 above)
Comment (eylles) [reply wireless.lua]: they are spaces now

File: wireless.lua (hunk 1 above)
Comment (eylles) [reply wireless.lua:184]: converted to spaces now 👍

**Comment by eylles** [comment]
please tell me if there's something else to correct, i'm rather new to lua.

**Comment by pltanton** [comment]
Looks fine! Thanks!

**Pull request merged**